src/ILLink.RoslynAnalyzer/RequiresUnreferencedCodeAnalyzer.cs

```diff
@@ -67,6 +72,23 @@ public override void Initialize (AnalysisContext context)
 						CheckMethodOrCtorCall (operationContext, prop.SetMethod);
 				}, OperationKind.PropertyReference);
 
+				static void CheckStaticConstructors (OperationAnalysisContext operationContext,
+					ImmutableArray<IMethodSymbol> constructors)
+				{
+					foreach (var constructor in constructors) {
+						if (constructor.Parameters.Length == 0 && constructor.HasAttribute (RequiresUnreferencedCodeAttribute)) {
```

Comment:
Why is this limited to parameterless cctors?

Reply:
Static constructors in C# cannot have access modifiers or have parameters
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/static-constructors

Reply:
Wouldn't a normal parameterless constructor satisfy this condition as well? Seems like it's working because we always pass `INamedTypeSymbol.StaticConstructors` (which is fine), shouldn't as well verify `MethodKind` or maybe that the md name is `cctor`?